packages/shopify-codemod/transforms/split-return-assignments.js

```diff
@@ -0,0 +1,112 @@
+import {matchLast} from './utils';
+
+export default function splitReturnAssignments({source}, {jscodeshift: j}, {printOptions = {}}) {
+  const sourceAST = j(source);
+  sourceAST
+    .find(j.Function, {
+      body: {
+        body: matchLast(j.ReturnStatement, {
+          type: 'ReturnStatement',
+          argument: {
+            type: 'AssignmentExpression',
+          },
+        }),
+      },
+    })
+    .forEach(({node: {body: {body}}}) => {
+      const {left, right} = body[body.length - 1].argument;
+      const returnLine = body[body.length - 1];
+      if (returnLine.argument.left) {
+        delete body[body.length - 1];
+        body.push({
+          type: 'ExpressionStatement',
+          expression: {
+            type: 'AssignmentExpression',
+            operator: '=',
+            left,
+            right,
+          },
+        });
+
+        if (returnLine.argument.left.name) {
+          body.push({
+            type: 'ReturnStatement',
+            argument: {
+              type: 'Identifier',
+              name: returnLine.argument.left.name,
+            },
+          });
+        } else {
+          body.push({
+            type: 'ReturnStatement',
+            computed: false,
+            argument: {
+              type: 'MemberExpression',
+              object: returnLine.argument.left.object,
+              property: returnLine.argument.left.property,
+            },
+          });
+        }
+      } else {
+        body.push({
+          type: 'ExpressionStatement',
+          expression: {
+            type: 'AssignmentExpression',
+            operator: '=',
+            left: {
+              type: 'MemberExpression',
+              object: returnLine.argument.object,
+              property: returnLine.argument.property,
+            },
+            right: {
+              type: 'MemberExpression',
+              object: returnLine.argument.object,
+              property: returnLine.argument.property,
+            },
+          },
+        });
+      }
+    });
+  sourceAST
+    .find(j.ExpressionStatement, {
+      type: 'ExpressionStatement',
+      expression: {
+        type: 'CallExpression',
```

Comment:
Still not resolved, so this will still not work for something like `const foo = () => bar = baz;`